src/domain/synthetics/leaderboards/usePositionsInfo.ts

```diff
@@ -0,0 +1,276 @@
+
+    const remainingCollateralUsd = collateralUsd.sub(totalPendingFeesUsd);
+
+    const remainingCollateralAmount = convertToTokenAmount(
+      remainingCollateralUsd,
+      collateralToken.decimals,
+      collateralMinPrice
+    )!;
+
+    const pnl = getPositionPnlUsd({
+      marketInfo: marketInfo,
+      sizeInUsd: position.sizeInUsd,
+      sizeInTokens: position.sizeInTokens,
+      markPrice,
+      isLong: position.isLong,
+    });
+
+    const pnlPercentage =
+      collateralUsd && !collateralUsd.eq(0) ? getBasisPoints(pnl, collateralUsd) : BigNumber.from(0);
+
+    const netValue = getPositionNetValue({
+      collateralUsd: collateralUsd,
+      pnl,
+      pendingBorrowingFeesUsd: position.pendingBorrowingFeesUsd,
+      pendingFundingFeesUsd: pendingFundingFeesUsd,
+      closingFeeUsd,
+    });
+
+    const pnlAfterFees = pnl.sub(totalPendingFeesUsd).sub(closingFeeUsd);
+    const pnlAfterFeesPercentage = !collateralUsd.eq(0)
+      ? getBasisPoints(pnlAfterFees, collateralUsd.add(closingFeeUsd))
+      : BigNumber.from(0);
+
+    const leverage = getLeverage({
+      sizeInUsd: position.sizeInUsd,
+      collateralUsd: collateralUsd,
+      pnl: showPnlInLeverage ? pnl : undefined,
+      pendingBorrowingFeesUsd: position.pendingBorrowingFeesUsd,
+      pendingFundingFeesUsd: pendingFundingFeesUsd,
+    });
+
+    const hasLowCollateral = leverage?.gt(MAX_ALLOWED_LEVERAGE) || false;
+
+    const liquidationPrice = getLiquidationPrice({
+      marketInfo,
+      collateralToken,
+      sizeInUsd: position.sizeInUsd,
+      sizeInTokens: position.sizeInTokens,
+      collateralUsd,
+      collateralAmount: position.collateralAmount,
+      userReferralInfo: undefined,
+      minCollateralUsd,
+      pendingBorrowingFeesUsd: position.pendingBorrowingFeesUsd,
+      pendingFundingFeesUsd,
+      isLong: position.isLong,
+    });
+
+    positionsMap[positionKey] = {
+      ...position,
+      marketInfo,
+      indexToken,
+      collateralToken,
+      pnlToken,
+      markPrice,
+      entryPrice,
+      liquidationPrice,
+      collateralUsd,
+      remainingCollateralUsd,
+      remainingCollateralAmount,
+      hasLowCollateral,
+      leverage,
+      pnl,
+      pnlPercentage,
+      pnlAfterFees,
+      pnlAfterFeesPercentage,
+      netValue,
+      closingFeeUsd,
+      pendingFundingFeesUsd,
+      pendingClaimableFundingFeesUsd,
+    };
+
+    return positionsMap;
+  },
+  {} as PositionsData);
+};
+
+export function usePositionsInfo(
+  chainId: number,
+  positionsHash: string,
+  positionKeys: string[],
+  marketPrices: ContractMarketPrices[],
+): PositionsResult {
+  const { minCollateralUsd } = usePositionsConstants(chainId);
+  const { marketsInfoData, tokensData, pricesUpdatedAt } = useMarketsInfo(chainId);
+  const [positions, setPositions] = useState<PositionsInfoData>();
+  const positionsData = useMulticall(chainId, "usePositionsData", {
```

Comment:
we have different usePositionsData multicall hook. the name "usePositionsData" is a part of SWR key. it would be better to specify unique name to avoid possible collisions